app/Jobs/ApplicationDeploymentJob.php

```
@@ -664,8 +664,14 @@ private function deploy_docker_compose_buildpack()
             // Always add .env file to services
             $services = collect(data_get($composeFile, 'services', []));
-            $services = $services->map(function ($service, $name) {
-                $service['env_file'] = ['.env'];
+            $isSelectiveEnabled = $this->application->settings->is_selective_env_per_service_enabled;
+
+            $services = $services->map(function ($service, $name) use ($isSelectiveEnabled) {
+                if ($isSelectiveEnabled) {
+                    $service['env_file'] = array_merge(data_get($service, 'env_file', []), [".env.$name"]);
+                } else {
+                    $service['env_file'] = array_merge(data_get($service, 'env_file', []), ['.env']);
+                }
```

PHPMD (2.15.0)

[error] The method deploy_docker_compose_buildpack uses an else expression. Else clauses are basically not necessary and you can simplify the code by not using them. (undefined)

(ElseExpression)